lib/gadget/gadget.vala

```diff
@@ -398,8 +398,12 @@ namespace Frida.Gadget {
 		}
 
 		public async void start () throws Error {
+			string listen_address = GLib.Environment.get_variable ("FRIDA_GADGET_LISTEN_ADDRESS");
```

Comment:
- This should be done at the higher level, and passed into the constructor instead – ordered first, just before `jit_enabled`. It could be a `string?`, and if `null` is passed in it will use `LISTEN_ADDRESS` instead.
- We should parse the address (and port if specified) instead of expecting the user to provide a cryptic URI string. Feel free to steal the code from `frida-server`: https://github.com/frida/frida-core/blob/3b3bc6c4779eca14f3c9cdf48c45f28f5b3944d5/server/server.vala#L39-L49